iiif_prezi3/helpers/add_thumbnail.py

```diff
@@ -21,5 +22,66 @@ def add_thumbnail(self, image_url, **kwargs):
         self.thumbnail.append(new_thumbnail)
         return new_thumbnail
 
+    def create_thumbnail_from_iiif(self, url, preferred_width=500, **kwargs):
+        """Adds an image thumbnail to a manifest or canvas based on a IIIF service.
+
+        Args:
+            url (str): An HTTP URL which points at a IIIF Image response.
+            preferred_width (int, optional): the preferred width of the thumbnail.
+            **kwargs (): see ResourceItem.
+
+        Returns:
+            new_thumbnail (ResourceItem): The updated list of thumbnails, including the newly-created one.
+        """
+        image_response = ResourceItem(id=url, type='Image')
+        image_info = image_response.set_hwd_from_iiif(url)
+        context = image_info.get('@context', '')
+
+        if 'sizes' in image_info:
+            best_fit_size = min(
+                image_info['sizes'], key=lambda size: abs(size["width"] - preferred_width)
+            )
+            thumbnail_id = f"{url.replace('/info.json', '')}/full/{best_fit_size['width']},{best_fit_size['height']}/0/default.jpg"
```

Comment:
This is iiif version dependant. For v3 the canonical form of parameters has changed. For v2 its /{width},/ and for v3 its /{width},{height}/:
https://iiif.io/api/image/3.0/change-log/#113-change-canonical-form-of-size-parameter-to-wh

Reply:
On line 63, I've added a conditional to override this if it's level0, has sizes and image v2.  I admit this is probably not the cleanest way to handle this.

Reply:
On line 63 I've added a conditional to override this if the response is image v2, level 0, and there are sizes.  Admittedly, this is probably not the cleanest way to address.